config/phinx.php

```diff
@@ -58,11 +56,23 @@
     }
 }
 
-return
-[
+$dataSource = 'default';
+$pluginConfig = sprintf('%1$s%2$sConfig%2$sphinx.php', $migrationRoot, DS);
+if (file_exists($pluginConfig)) {
+    require_once $pluginConfig;
+    if (null !== Configure::read('phinx.datasource')) {
+        $dataSource = Configure::read('phinx.datasource');
```

Comment:
Because `Configure` is a de-facto global class, the configuration `phinx.datasource` may be set in one plugin, having an effect on another. Example: Plugins `A` and `B` both have a `Config/Migrations` directory containing migrations, but in Plugin `B` the developer has a typo in `Config/phinx.php`: `Configure::write('phinx.datasorce', 'plugin_b_database');`. The migrations from plugin `B` will be applied to the database of plugin `A`, because `Configure::read('phinx.datasource')` still returns returns the value set by plugin `A`.

I think it's better to use `Configure::consume()` for cases of one-off configurations:

```
/**
 * Used to read and delete a variable from Configure.
 *
 * This is primarily used during bootstrapping to move configuration data
 * out of configure into the various other classes in CakePHP.
 *
 * @param string $var The key to read and remove.
 * @return array|null
 */
```

Reply:
Each Plugin will be executed in its own call.

e.g.
```bash
PLUGIN=<plugin_name> ./vendor/bin/phinx-cake2.sh migrate
```
Therefor no side-effects will happen as mentioned.